src/zcl/zclHeader.ts

```diff
@@ -1,10 +1,51 @@
 import {FrameControl} from './definition';
+import BuffaloZcl from './buffaloZcl';
 
-interface ZclHeader {
-    frameControl: FrameControl;
-    manufacturerCode: number;
-    transactionSequenceNumber: number;
-    commandIdentifier: number;
+const MINIMAL_HEADER_LENGTH = 3;
+
+class ZclHeader {
+    public readonly frameControl: FrameControl;
+    public readonly manufacturerCode: number;
+    public readonly transactionSequenceNumber: number;
+    public readonly commandIdentifier: number;
+
+    constructor(frameControl: FrameControl, manufacturerCode: number, transactionSequenceNumber: number, commandIdentifier: number) {
+        this.frameControl = frameControl;
+        this.manufacturerCode = manufacturerCode;
+        this.transactionSequenceNumber = transactionSequenceNumber;
+        this.commandIdentifier = commandIdentifier;
+    }
+
+    public getLength(): number {
+        // Returns the amount of bytes of this header
+        return 3 + (this.manufacturerCode === null ? 0 : 2);
+    }
+
+    public static fromBuffer(buffer: Buffer): ZclHeader {
```

Comment:
Can this return `ZclHeader | undefined` instead of throwing? So in adapters code:

```typescript
const payload: ZclDataPayload: {
    header: ZclHeader.fromBuffer(messageContents),
    ...
};
```

Controller can log if `undefined`.

Reply:
Good idea, will also do this once all tests are fixed.

Reply:
Checking this now, just `throw new Error("ZclHeader length is lower than minimal length");`  -> `return undefined` is not enough. It can also go wrong at other lines in this function. Which mean we would have to do:

```js
try {
 .. all function code
} catch (error) {
  return undefined
}
```

which doesn't feel right to me.